Purchases/Networking/HTTPRequest.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  HTTPRequest.swift
+//
+//  Created by Nacho Soto on 2/27/22.
+
+/// A request to be made by ``HTTPClient``
+struct HTTPRequest {
+
+    let method: Method
+    let path: Path
+
+}
+
+// MARK: - Method
+
+extension HTTPRequest {
+
+    typealias Body = [String: Any]
+
+    enum Method {
+
+        case get
+        case post(body: Body)
+
+    }
+
+}
+
+extension HTTPRequest {
+
+    var requestBody: Body? {
+        switch self.method {
+        case let .post(body): return body
+        case .get: return nil
+        }
+    }
+
+}
+
+extension HTTPRequest.Method {
+
+    var httpMethod: String {
+        switch self {
+        case .get: return "GET"
+        case .post: return "POST"
+        }
+    }
+
+}
+
+// MARK: - Path
+
+extension HTTPRequest {
+
+    enum Path {
+
+        case getCustomerInfo(appUserID: String)
+        case getOfferings(appUserID: String)
+        case getIntroEligibility(appUserID: String)
+        case logIn
+        case createAlias(appUserID: String)
+        case postAttributionData(appUserID: String)
+        case postOfferForSigning
+        case postReceiptData
+        case postSubscriberAttributes(appUserID: String)
+
+    }
+
+}
+
+extension HTTPRequest.Path: Hashable {}
+
+extension HTTPRequest.Path: CustomStringConvertible {
```

Comment:
even though this is a very simple type, I kinda feel like we should have tests for it, since we're no longer hardcoding the paths in BackendTests anymore.
My worry is that we could have a typo or miss a `/` and we'd have no tests against it

Reply:
Other than integration tests, but yeah I'll add that!

Reply:
Done: [69e9e20](https://github.com/RevenueCat/purchases-ios/pull/1347/commits/69e9e200b511da1abe86c051082d572e90a52ddf)